Optimizations.ipynb

```diff
@@ -0,0 +1,1351 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "20af2285",
```

Comment:
whats the diff in the 2 notebooks?

Reply:
https://docs.google.com/document/d/13kgl_4QY2T9ODUrLtasu9MxsyAUduMjZT5W6Rg0m190/edit?pli=1